### PR TITLE
added a feature flip to flip the feature flip to the new api

### PR DESCRIPTION
**Pull request by jspies**
This PR makes the flippy gem able to go directly to Redis or to the new API depending on the value of flippy_use_api in feature flip 1.

This is intended as a transition codebase and should be use to stabilize feature flip 2 in production and then be removed.

https://bypassmobile.atlassian.net/browse/DEV-8139

**Deployment Considerations**

This gem should be considered deployable to monolith and admin once tested as it should enable the ability to decouple deployment from release. Once the new feature flip flow is verified, the flip can be turned on and everything tested. Once running stable, the next update should remove the old store.

### CODE REVIEW

**Comment by dvictory** [comment]
1. What is the data store is being used by http://feature-flip.bypassmobile.com? Both? ie. What happens if you are not using the Gem to talk to feature flip?  
2. How do you turn on and off feature flip 2 without breaking things that have been enabled/disabled since switching.

**Comment by teubanks** [comment]
It's not surprising that it's a little confusing. Basically, the old code that talked directly to Redis was all done via class methods. The new code, that talks to the feature-flip api, is all instance methods. We check Redis to see if we should be talking to the old or the new (see [here](https://github.com/bypasslane/flippy/pull/3/files#diff-7e0cf2802c5dc7cc32f92680aed1f4bcR15) and [here](https://github.com/bypasslane/flippy/pull/3/files#diff-7e0cf2802c5dc7cc32f92680aed1f4bcR29) for where the magic happens)

**Comment by jspies** [comment]
This gem will always rely on the old feature flip to check the initial feature flip. So the system will be dependent on both systems. Once we verify the stability, we will remove the dependency.

File: lib/flippy/floppy.rb [hunk 1]
```diff
@@ -8,66 +8,93 @@
...
   end
 
-  def flippy_floppy(key, scope: {}, session_token:)
-    flipFlopClient.feature(key: key, scope: scope, session_token: session_token)
+  def flippy_floppy(key, scope: {}, session_token: nil)
+    if Store.check(key: "flippy_use_api")
+      raise ArgumentError.new("session_token required") unless session_token
+      flipFlopClient.feature(key: key, scope: scope, session_token: session_token)
+    else
+      Flippy::Floppy.feature(key: key, scope: scope)
```
Comment (teubanks) [inline lib/flippy/floppy.rb]: Should we start throwing a warning here to let people know they're calling `flippy_floppy` without passing a session token? That might help get ahead of the curve on people not passing one

Comment (jspies) [inline lib/flippy/floppy.rb]: that's a good idea. can you add it?

**Comment by jspies** [comment]
@teubanks very nice, i'm good with this PR

**Pull request merged**